src/dstack/_internal/core/models/configurations.py

```diff
@@ -143,6 +150,14 @@ def convert_python(cls, v, values) -> Optional[PythonVersion]:
             return PythonVersion(v)
         return v
 
+    @validator("volumes", each_item=True)
+    def convert_volumes(cls, v) -> Union[VolumeMountPoint, InstanceMountPoint]:
+        if isinstance(v, str):
+            if v.startswith("/"):
+                return InstanceMountPoint.parse(v)
+            return VolumeMountPoint.parse(v)
```

Comment:
So now we support `volume-name:mount_path`? Please mention such things in the PR description or in the docs.

Reply:
Is it a good idea at all? User may try to use relative paths for instance mounts but will get an error about volumes?

Reply:
I've improved mount point type detection, if the left part contains a slash in any position, we are sure it's an instance mount point: https://github.com/dstackai/dstack/pull/1895/commits/8e0396e37ac5291dcc8a71469bb49053773c49e9#diff-6b2463188801b3ca761b5edf2026af9f3c226b150fb67b5c7fad1665bdeacc9eR96-R99

The only case when the ambiguity is still present is when the user tries to mount a relative instance path `foo` with the `foo:/path/to` syntax — we assume it's a volume mount and fail with `Volume foo not found`.

Not sure if the verbose message [here](https://github.com/dstackai/dstack/blob/d8b11a3381a3d0a9073eb95843e402acc1a405ca/src/dstack/_internal/server/services/runs.py#L635) (something like “to use a network volume {name_or_rel_path}, create it with the volume configuration, to mount a local path, specify an absolute path `/{name_or_rel_path}`”) would greatly improve user experience, for example, it the user uses the full form syntax:

```yaml
volumes:
  - name: foo
    path: /volume
```

it's clear that they tries to mount a network volume, but the error message would still suggest a local volume fix as well. What do you think?

Reply:
> we assume it's a volume mount and fail with 

I think it's ok. No need for a different error message.